Mathlib/Algebra/BigOperators/Order.lean

```diff
@@ -816,3 +816,61 @@ theorem IsAbsoluteValue.map_prod [CommSemiring R] [Nontrivial R] [LinearOrderedC
 #align is_absolute_value.map_prod IsAbsoluteValue.map_prod
 
 end AbsoluteValue
+
+namespace Mathlib.Meta.Positivity
+open Qq Lean Meta Finset
+
+/-- Prove that a finset is nonempty from a local assumption or from the fact that the finset is
+`univ` in a nonempty fintype.
+
+Note that this does not do any complicated reasoning: the only ways it can prove `s` is nonempty is
+if there is a local `s.Nonempty` hypothesis or `s = (univ : Finset α)` and `Nonempty α` can be
+synthesized by TC inference.
+
+TODO: Replace by an aesop ruleset? -/
```

Comment:
I tried for a bit to get this to work, but no success. Here's a Zulip thread: https://leanprover.zulipchat.com/#narrow/stream/239415-metaprogramming-.2F-tactics/topic/Calling.20aesop.20from.20a.20tactic.20to.20obtain.20an.20.60Expr.60